langchain/vectorstores/matching_engine.py

```diff
@@ -0,0 +1,473 @@
+        # and the similarity_search method only recevies one query. This
+        # means that the match method will always return an array with only
+        # one element.
+        for doc in response[0]:
+            page_content = self._download_from_gcs(f"documents/{doc.id}")
+            results.append(Document(page_content=page_content))
+
+        logger.debug(f"Downloaded documents for query.")
+            
+        return results
+
+    def _get_index_id(self) -> str:
+        """Gets the correct index id for the endpoint.
+
+        Returns:
+            The index id if found (which should be found) or throws
+            ValueError otherwise.
+        """
+        for index in self.endpoint.deployed_indexes:
+            if index.index == self.index.resource_name:
+                return index.id
+
+        raise ValueError(f"No index with id {self.index.resource_name} "
+                         f"deployed on enpoint "
+                         f"{self.endpoint.display_name}.")
+
+    def _download_from_gcs(self, gcs_location: str) -> str:
+        """Downloads from GCS in text format.
+
+        Args:
+            gcs_location: The location where the file is located.
+
+        Returns:
+            The string contents of the file.
+        """
+        client = self._get_gcs_client()
+        bucket = client.get_bucket(self.gcs_bucket_name)
+        blob = bucket.blob(gcs_location)
+        return blob.download_as_string()
+    
+    @classmethod
+    def from_texts(
+        cls: Type["MatchingEngine"], 
+        texts: List[str],
+        embedding: Embeddings = None,
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> "MatchingEngine":
+        """Return VectorStore initialized from texts and embeddings.
+
+        Note that this function shouldn't be run more than once. Otherwise,
+        the texts will be added multiple times to the index and GCS.
+
+        Args:
+            texts: The texts that will get .
+            embedding: The :class:`Embeddings` that will be used for
+            embedding the texts.
+            metadatas: List of metadatas. Defaults to None.
+
+            Required kwargs:
+                project_id: The GCP project id.
+                region: The default location making the API calls. It must have
+                the same location as the GCS bucket and must be regional.
+                gcs_bucket_uri: The location where the vectors will be stored in
+                order for the index to be created.
+                index_id: The id of the created index.
+                endpoint_id: The id of the created endpoint.
+                credentials_path: The path of the Google credentials on the
+                local file system.
+
+        Returns:
+            A configured MatchingEngine with the texts added to the index.
+        """
+        required_kwargs = ["project_id", "region", "gcs_bucket_uri",
```

Comment:
If these are required, it's better to make them explicit. It seems better to do:
```python
    @classmethod
    def from_texts(
        cls: Type["MatchingEngine"], 
        texts: List[str],
        embedding: Embeddings = None,
        metadatas: Optional[List[dict]] = None,
        *,
        project_id: str,
        region: str,
        gcs_bucket_name: str,
...
```
etc.

Also - we should unify `gcs_bucket_uri` since it was schanged to `gcs_bucket_name` in the init (same as below)

If we're doing required kwargs here, should do required kwargs in from_components as well.


        **kwargs: Any,
    ) -> "MatchingEngine":